clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/names/EcsServerGroupNameResolver.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.ecs.names;
+
+import com.amazonaws.services.ecs.AmazonECS;
+import com.amazonaws.services.ecs.model.*;
+import com.google.common.collect.Lists;
+import com.netflix.spinnaker.moniker.Moniker;
+import com.netflix.spinnaker.moniker.Namer;
+import java.util.*;
+import java.util.stream.IntStream;
+
+public class EcsServerGroupNameResolver {
+
+  private static final int SEQUENTIAL_NUMBERING_NAMESPACE_SIZE = 1000;
+
+  private static final int MAX_NEXT_SERVER_GROUP_ATTEMPTS = 5;
+
+  private final String ecsClusterName;
+  private final AmazonECS ecs;
+  private final String region;
+  private final Namer<EcsResource> naming;
+
+  public EcsServerGroupNameResolver(
+      String ecsClusterName, AmazonECS ecs, String region, Namer<EcsResource> namer) {
+    this.ecsClusterName = ecsClusterName;
+    this.ecs = ecs;
+    this.region = region;
+    this.naming = namer;
+  }
+
+  public EcsServerGroupName resolveNextName(String application, String stack, String detail) {
+    Moniker m = Moniker.builder().app(application).detail(detail).stack(stack).build();
```

Comment:
nit: can you rename the variable to something more descriptive than `m` possibly something like `moniker`?